.github/workflows/ci.yml

```diff
@@ -0,0 +1,43 @@
+# This workflow will install Python dependencies, runs lint, static typechecking and tests
+# with multiple versions of Python. 
+
+name: Continuous Integration
+
+on: push
```

Comment:
Do we want to run this on every push?

I think we should limit it to pull requests for master. Depends on what the purpose of this workflow would be at the end. If the intention is to use this to check for consistency, formatting and passing tests, then PR to main/master would be sufficient to trigger this imo

Reply:
Actions on open source repos are free, so I would recommend running them on every push. So you can see early if the CI got messed up before merging and it only takes a few minutes. Or do I miss some fundamental concepts here.

Reply:
The running on PR only will achieve the same purpose. Any branch that submits a PR to main/master will run the CI. 

We can also run it on push for any branch but the downside might be that there might be quite a few commits pushed to a branch before a PR is made ready (In personal experience at least) and then we don't really need to run the CI (once we have more comprehensive testing) on each such commit as the purpose is to only check if everything looks okay before merging.